// @osd/analytics/target/node/metrics/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApplicationUsage", {
  enumerable: true,
  get: function () {
    return _application_usage.ApplicationUsage;
  }
});
Object.defineProperty(exports, "ApplicationUsageCurrent", {
  enumerable: true,
  get: function () {
    return _application_usage.ApplicationUsageCurrent;
  }
});
exports.METRIC_TYPE = void 0;
Object.defineProperty(exports, "Stats", {
  enumerable: true,
  get: function () {
    return _stats.Stats;
  }
});
Object.defineProperty(exports, "UiStatsMetric", {
  enumerable: true,
  get: function () {
    return _ui_stats.UiStatsMetric;
  }
});
Object.defineProperty(exports, "UiStatsMetricType", {
  enumerable: true,
  get: function () {
    return _ui_stats.UiStatsMetricType;
  }
});
Object.defineProperty(exports, "createUiStatsMetric", {
  enumerable: true,
  get: function () {
    return _ui_stats.createUiStatsMetric;
  }
});
Object.defineProperty(exports, "trackUsageAgent", {
  enumerable: true,
  get: function () {
    return _user_agent.trackUsageAgent;
  }
});
var _ui_stats = require("./ui_stats");
var _stats = require("./stats");
var _user_agent = require("./user_agent");
var _application_usage = require("./application_usage");
/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Any modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */
/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
let METRIC_TYPE = exports.METRIC_TYPE = /*#__PURE__*/function (METRIC_TYPE) {
  METRIC_TYPE["COUNT"] = "count";
  METRIC_TYPE["LOADED"] = "loaded";
  METRIC_TYPE["CLICK"] = "click";
  METRIC_TYPE["USER_AGENT"] = "user_agent";
  METRIC_TYPE["APPLICATION_USAGE"] = "application_usage";
  return METRIC_TYPE;
}({});
